### ISSUE TITLE
Kubernetes Service: Docs do not describe Kubernetes API requirements

### ISSUE THREAD

**Issue by vaheed** [apache/cloudstack#6437]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Kubernetes Service
~~~

##### CLOUDSTACK VERSION

~~~
4.16.1.0 LTS
~~~

##### CONFIGURATION

Advanced zone without Security Profile

##### OS / ENVIRONMENT

Ubuntu 20.04 OS on KVM hypervisor

##### SUMMARY

Unable to create new Kubernetes cluster
I was able to register K8s from the original ACS repo, I was able to add a new supported Kubernetes version but was unable to create a cluster as its throwing exception

##### STEPS TO REPRODUCE

~~~
1. Enable the Kubernetes service
2. Download the iso from http://download.cloudstack.org/cks/
3. Register a new Kubernetes version using
mincpunumber=2 minmemory=2048
4. Wait till its ready
5. Now create a Kubernetes cluster from UI with 2 nodes
~~~

##### EXPECTED RESULTS

~~~
Expected Kubernetes cluster to be created successfully
~~~

##### ACTUAL RESULTS

~~~
The cluster status is stuck in "Starting" forever and keeps throwing an exception
The VMs are created successfully
~~~
~~~
2022-06-05 04:59:55,946 WARN [o.a.c.f.j.i.AsyncJobMonitor] (Timer-0:ctx-8669d7f6) (logid:933baacc) Task (job-326) has been pending for 1083 seconds
2022-06-05 04:59:59,472 WARN [c.c.k.c.u.KubernetesClusterUtil] (API-Job-Executor-7:ctx-c7dbf788 job-326 ctx-2117db03) (logid:ee321178) API endpoint for Kubernetes cluster : k8s011 not available 
javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake Caused by: java.io.EOFException: SSL peer shut down incorrectly
~~~

This is the result from ACS management when trying to k8s master node curl -k https://172.21.21.108:6443/version
~~~
curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to 172.21.21.108:6443
~~~

**Comment by zap51** [comment]
Hello @vaheed ,
    Looks like your issue is related to https://github.com/apache/cloudstack/issues/5999 and one manual fix could be https://github.com/apache/cloudstack/issues/5999#issuecomment-1052117682. This is however fixed on PR https://github.com/apache/cloudstack/pull/6076 and the latest release of CloudStack 4.17.0.0 has no issues.

For the Kubernetes network requirements, please ensure network connectivity as per https://github.com/apache/cloudstack/issues/4146#issuecomment-720327587 for the nodes to get ready. In your case, perhaps the Kubernetes Master Node(s) would have been experiencing issue described in https://github.com/apache/cloudstack/issues/5999.

**Comment by vaheed** [comment]
Thanks, @zap51, fixed in 4.17 :)
